pkg/policy/validate.go

```diff
@@ -50,6 +82,11 @@ func Validate(policy *kyverno.ClusterPolicy, client *dclient.Client, mock bool,
 	}
 
 	for i, rule := range p.Spec.Rules {
+		//check for forward slash
+		if JSONPatchPathHasForwardSlash(rule) != true {
```

Comment:
Here you are passing complete rule. 
Consider the following policy :
```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: policy-remove-label
spec:
  rules:
    - name: "Remove unwanted label"
      match:
        resources:
          kinds:
          - Secret
      mutate:
        patchesJson6902: |-
          - path: "/metadata/labels/purpose"
            op: remove          
```
here you need to pass only path i.e., ` - path: "/metadata/labels/purpose"`

Reply:
>Here you are passing complete rule.

yeah I understand that 
while adding this PR  what I thought was that 
using `if JSONPatchPathHasForwardSlash(rule)` we are passing the whole rule and then using the function  ` JSONPatchPathHasForwardSlash()` we are checking for the forward slash in the path
```
func JSONPatchPathHasForwardSlash(patch string) bool {

	jsonPatch, err := yaml.ToJSON([]byte(patch))
	if err != nil {
		return err
	}

	decodedPatch, err := jsonpatch.DecodePatch(jsonPatch)
	if err != nil {
		return err
	}

	for _, operation := range decodedPatch {
		path, err := operation.Path()
		if err != nil {
			return err
		}

		out, err := regexp.MatchString(`^/`, path)
		if err != nil {
			return fmt.Errorf("Path :", err)
		}

		return out
	}
	return nil
}
```


I know that the function is incorrect 

As you suggested that we need to check for the path in the for loop but I am not able to figure out how it needs to be done

Reply:
First point, the function `JSONPatchPathHasForwardSlash` takes a string parameter but you are passing `rule` which is of type `kyverno.Rule`. Here pass only `PatchesJSON6902` and extract the required string.

Here is the rule struct:
https://github.com/kyverno/kyverno/blob/c522343c038ba8dea652137cbcdc2ae1f3a0bf29/pkg/api/kyverno/v1/policy_types.go#L62

Here what you need to find in rule and pass:
https://github.com/kyverno/kyverno/blob/c522343c038ba8dea652137cbcdc2ae1f3a0bf29/pkg/api/kyverno/v1/policy_types.go#L367

When you will pass `PatchesJSON6902` print and check what you are getting and what you are suppose to extract from the complete string.